### PR TITLE
Dockerize frontend + backend

### PR DESCRIPTION
**Pull request by ekweirui**
This pull request aims to do the following:
1) Accomplish milestone 3 by containerizing the entire application (frontend + backend)
2) Adopt the microservices architecture in the backend

### CODE REVIEW

File: frontend/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+RUN npm install
+
+COPY . .
+
+RUN npm run build
+
+FROM nginx:alpine
+
+ COPY nginx.conf /etc/nginx/nginx.conf
```
Comment (Tan-Jia-Rong) [inline frontend/Dockerfile]: why got spacebar in front

**Review by Tan-Jia-Rong — APPROVED** [review]
LGTM 💯

File: frontend/Dockerfile (hunk 1 above)
Comment (ekweirui) [reply frontend/Dockerfile]: Good catch! Updated it 😄

**Review by Tan-Jia-Rong — APPROVED** [review]
LGTM thanks for dockerizing the files :>

**Pull request merged**